onnxruntime/core/graph/model.cc

```diff
@@ -447,7 +448,13 @@ Status Model::Load(int fd, ONNX_NAMESPACE::ModelProto& model_proto) {
   }
 
 #if GOOGLE_PROTOBUF_VERSION >= 3002000
-  FileInputStream input(fd);
+  size_t file_size = 0;
+  int block_size = -1;
+  Status st = Env::Default().GetFileLength(fd, file_size);
+  if (st.IsOK()) {
+    block_size = static_cast<int>(file_size);
```

Comment:
What if the file is 1GB, or even more than 2GB?

Reply:
How about set the block size to std::min(4*1024*1024, file_size)? In most of the cases, model file is larger than 4MB, and setting a such large buffer size doesn't bring much benefit I guess.

Reply:
One drawback of this approach this GetFileLength only work with regular files, not pipes.

Reply:
Using std::min is fine. Is the 4mb number based on some experimentation?
Do we've any requirement to make this work on pipes?